src/main/java/ch/njol/skript/util/slot/EquipmentSlot.java

```diff
@@ -180,9 +181,18 @@ public ItemStack getItem() {
 	
 	@Override
 	public void setItem(final @Nullable ItemStack item) {
-		slot.set(e, item);
-		if (e.getHolder() instanceof Player)
-			PlayerUtils.updateInventory((Player) e.getHolder());
+		// Small bandaid for Spigot's AssertionError "TRAP" (crashing the server) when setting a player's tool to air/null during some events
+		// We add a small delay, to set the item to air/null after Minecraft has handled changing the item amount
+		if (e.getHolder() instanceof Player && (slot == EquipSlot.TOOL || slot == EquipSlot.OFF_HAND) && (item == null || item.getType() == Material.AIR)) {
```

Comment:
Double ``e.getHolder()`` check, the if statement could be simplified.

Reply:
Not really. 1 is used in a delayed task. 
But if you have a suggestion to simplify it, please let me know.

Reply:
```java
		// Small bandaid for Spigot's AssertionError "TRAP" (crashing the server) when setting a player's tool to air/null during some events
		// We add a small delay, to set the item to air/null after Minecraft has handled changing the item amount
		if (e.getHolder() instanceof Player) {
			if ((slot == EquipSlot.TOOL || slot == EquipSlot.OFF_HAND) && (item == null || item.getType() == Material.AIR)) {
				Bukkit.getScheduler().scheduleSyncDelayedTask(Skript.getInstance(), () -> {
					slot.set(e, null);
					PlayerUtils.updateInventory((Player) e.getHolder());
				});
			} else {
				slot.set(e, item);
				PlayerUtils.updateInventory((Player) e.getHolder());
			}
		} else {
			slot.set(e, item);
		}
```
Hm... not sure if this is any better than your code though.

Reply:
Yeah, doesn't really seem to improve it any.